### PR TITLE
Feat: Multiple votes per user, radius box, categories

### PR DESCRIPTION
**Pull request by joklek**
Some commits are from old branches, don't mind them.

* Each vote in the database now has a timestamp. Cooldown saved in WebApi/Web.config "ratingCooldownPeriod", currently at 15 minutes.
* All fetched bars are saved to our database. Earlier we would save only the ones that users have voted on.
* If user tries to vote when cooldown hasn't ended, server sends response.
* Radius box implemented.
* Round average rating in web UI.
* Categories returned as enums are properly interpreted in web UI.
* Error messages properly shown to user.

### CODE REVIEW

File: WebApi/Controllers/DataController.cs [hunk 1]
```diff
@@ -58,34 +59,50 @@ public IHttpActionResult GetAllBarData([FromBody]UserAndBarsModel userAndBars)
...
-        public async Task<IHttpActionResult> GetBarsByCoordinates([FromBody]Coordinate coordinate)
+        public async Task<IHttpActionResult> GetBarsByCoordinates([FromBody] Coordinate coordinate)
         {
-            var locationRequest = new LocationRequestModel { Latitude = coordinate.lat, Longitude = coordinate.lng, Radius = 150, User =  "test"  };
+            var locationRequest = new LocationRequestModel
+            {
+                Latitude = coordinate.lat,
+                Longitude = coordinate.lng,
+                Radius = coordinate.radius,
+                User = "test"
```
Comment (DomantasDev) [inline WebApi/Controllers/DataController.cs]: Do we really need a user at this point?

Comment (joklek) [reply WebApi/Controllers/DataController.cs]: Thanks for pointing that out. Removed this method altogether, will use the one that uses User powers.

Or do you mean, that we don't need a username for bar fetching?

**Comment by DomantasDev** [comment]
So I see you implemented the User, but the question was, do we really need that? before on request for bars around we were sending the user so we could get his current rating for each bar, but since now there are multiple ratings on each bar from the same user, how are we using the user info we send on request?

**Comment by joklek** [comment]
Sending user info has two advantages for future development: we could easily disallow bar requests for unregistered users and we could track user locations. As currently neither is really an issue, we can stop sending user info as you suggested. Will fix in 5 hours.

**Comment by joklek** [comment]
Resolved

**Pull request merged**